### PR TITLE
Canvas: Add direction options for connections

### LINKED ISSUE #83268: Canvas: Arrow head options
Add options for arrow heads:

1. Forward
2. Reverse
3. Bi-directional
4. None (line)

Source / target style independent.

### PR DESCRIPTION
**Pull request by drew08t**
Adding direction option for canvas connections.

For single segment connection:
Mar-15-2024 16-08-17

For multi-segment connection:
Mar-15-2024 16-04-28

Closes #83268

### CODE REVIEW

File: public/app/plugins/panel/canvas/components/connections/ConnectionSVG.tsx
Comment (adela-almasan) [inline public/app/plugins/panel/canvas/components/connections/ConnectionSVG.tsx:226]: maybe we can extract the logic for `markerEnd`/`markerStart` since we're reusing it?  😁

**Review by adela-almasan — APPROVED** [review]
Tested locally and it works good! It should be good to merge after minor cleanup! 🚀

**Review by baldm0mma — APPROVED** [review]

File: public/app/plugins/panel/canvas/components/connections/ConnectionSVG.tsx
Comment (drew08t) [reply public/app/plugins/panel/canvas/components/connections/ConnectionSVG.tsx:151]: Good catch!

**Comment by lukasztyrala** [comment]
LGTM, we can work on the labels for selector (or using icons), but this is not something blocking this PR.

File: public/app/plugins/panel/canvas/editor/options.ts
Comment (codeincarnate) [inline public/app/plugins/panel/canvas/editor/options.ts:133]: Labels for the form fields should be capitalized 😸

File: public/app/plugins/panel/canvas/editor/options.ts
Comment (codeincarnate) [inline public/app/plugins/panel/canvas/editor/options.ts:126]: May make sense to call this "Connection Direction" or something like that. It's a bit ambiguous at the moment.

**Review by codeincarnate — APPROVED** [review]
Works as expected! Just had a few nits 😄

File: public/app/plugins/panel/canvas/editor/options.ts
Comment (drew08t) [reply public/app/plugins/panel/canvas/editor/options.ts:131]: I think in this case, because it's in the context of a selected connection, I'll leave it as is for now to be consistent with its siblings.
image

Comment (drew08t) [reply public/app/plugins/panel/canvas/editor/options.ts:131]: Or perhaps "Arrow Direction" would suffice?

File: public/app/plugins/panel/canvas/components/connections/ConnectionSVG.tsx
Comment (drew08t) [reply public/app/plugins/panel/canvas/components/connections/ConnectionSVG.tsx:226]: Good suggestion!

**Pull request merged**
